src/bot-handlers/tests/liquidator-test.js

```diff
@@ -0,0 +1,497 @@
+            // amount of collateral minus the collateral withdrawn. 125 - 10 = 115
+            const expectedAlerts = [];
+            financialContractClient.getAllPositions().filter(
+              // only sponsor1 should be liquidatable
+              (position) => position.sponsor === sponsor1,
+            ).forEach((position) => {
+              expectedAlerts.push(createAlert(financialContractClient, position, toWei('1')));
+            });
+
+            assert.deepEqual(await handleBlock(), expectedAlerts);
+          },
+        );
+
+        describe('Agent correctly identifies liquidatable funding rates from perpetual contract', () => {
+          versionedIt([{ contractType: 'Perpetual', contractVersion: '2.0.1' }])(
+            'Can correctly detect invalid positions',
+            async () => {
+              // sponsor1 creates a position with 125 units of collateral, creating 100 synthetic tokens.
+              await financialContract.methods
+                .create({ rawValue: convertDecimals('125') }, { rawValue: convertDecimals('100') })
+                .send({ from: sponsor1 });
+
+              // sponsor2 creates a position with 150 units of collateral, creating 100 synthetic tokens.
+              await financialContract.methods
+                .create({ rawValue: convertDecimals('150') }, { rawValue: convertDecimals('100') })
+                .send({ from: sponsor2 });
+
+              // Both token sponsors should still have their positions with full collateral.
+              assert.equal(
+                (await financialContract.methods.getCollateral(sponsor1).call()).rawValue,
+                convertDecimals('125'),
+              );
+              assert.equal(
+                (await financialContract.methods.getCollateral(sponsor2).call()).rawValue,
+                convertDecimals('150'),
+              );
+
+              // Start with a mocked price of 1 usd per token.
+              // This puts both sponsors over collateralized so no liquidations should occur.
+              priceFeedMock.setCurrentPrice(toWei('1'));
+
+              // ensure no findings are reported initially
+              assert.deepEqual(await handleBlock(), []);
+
+              // Next, introduce some funding rate. Setting the funding rate multiplier to 1.04, results in modifying
+              // sponsor's debt. This becomes 100*1.04 = 104. All this debt, with a price of 1, both sponsors are
+              // still correctly capatalized with sponsor1 @ 125 / (104 * 1) = 1.202 & sponsor2 @ 150 / (104 * 1) = 1.44.
+              // So, if there is 150 collateral backing 105 token debt, with a collateral requirement of 1.2, then
+              // the price must be <= 150 / 1.2 / 105 = 1.19. Any price above 1.19 will cause the dispute to fail.
+              await setFundingRateAndAdvanceTime(toWei('0.000004'), contractCreator);
+              priceFeedMock.setCurrentPrice(toWei('1'));
```

Comment:
Do you also need to call `.update()` on the `priceFeedMock` when you set new prices?

Reply:
```js
29    public setCurrentPrice(currentPrice: BN | string | number | null): void {
  1     // allows this to be set to null without throwing.
  2     this.currentPrice = currentPrice !== null ? toBN(currentPrice.toString()) : currentPrice;
  3   }
```
```js
  2   public getCurrentPrice(): BN | null {
  1     return this.currentPrice || null;
61    }
```
no